diff --git a/include/tracefs-local.h b/include/tracefs-local.h
index ffc9d33..215bea0 100644
--- a/include/tracefs-local.h
+++ b/include/tracefs-local.h
@@ -9,7 +9,7 @@
 #include <tracefs.h>
 #include <pthread.h>
 
-#define __hidden __attribute__((visibility ("hidden")))
+#define __hidden __attribute__((weak, visibility ("hidden")))
 #define __weak __attribute__((weak))
 
 #define ARRAY_SIZE(array) (sizeof(array) / sizeof((array)[0]))
